drivers/power/pm/pm_runtime.c

```diff
@@ -0,0 +1,322 @@
+/****************************************************************************
+ * drivers/power/pm/pm_runtime.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <debug.h>
+#include <nuttx/clock.h>
+#include <nuttx/power/pm_runtime.h>
+
+#include "pm.h"
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int rpm_changestate(FAR struct pm_runtime_s *rpm, rpm_state_e state);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int rpm_suspend(FAR struct pm_runtime_s *rpm)
+{
+  int ret = 0;
+
+  if (rpm->ops && rpm->ops->runtime_suspend)
```

Comment:
Please include an assertion or an if (rpm && rpm->...) to guarantee rpm is not NULL, otherwise rpm->ops could create a null pointer exception

Reply:
it is an internal function, let's add assert or check begin the public function.

Reply:
done

Reply:
@acassis  Please view again